darts/utils/model_selection.py

```diff
@@ -0,0 +1,79 @@
+from typing import Sequence, Optional, Union
+from ..timeseries import TimeSeries
+
+
+def train_test_split(
+        data: Sequence[TimeSeries],
+        test_size: Optional[Union[float, int]] = 0.25,
+        axis: Optional[int] = 0,
+        n: Optional[int] = 0,
+        horizon: Optional[int] = 0,
+        ) -> Sequence[Sequence[TimeSeries]]:
+
+    """
+    Splits the dataset into training and test dataset. Supports splitting along the sample axis and time axis.
+
+    Parameters
+    ----------
+    data
+        original dataset to split into training and test
+
+    test_size
+        size of the test set. If the value is between 0 and 1, parameter is treated as a split proportion. Otherwise
+        it is treated as a absolute number of samples from each timeseries that will be in the test set. [default = 0.25]
+
+    axis
+        Axis to split the dataset on. When 0 (default) it is split on samples. Otherwise, if axis = 1,
+        timeseries are split along time axis (columns).
+
+    n
+        size of the input
+
+    horizon
+        forecast horizon
+
+    Returns
+    -------
+    Sequence[TimeSeries], Sequence[TimeSeries]
+        Training and test datasets tuple.
+    """
+
+    # TODO: support splitting covatiates at the same time
+
+    if axis == 0:
+
+        if 0 < test_size < 1:
+            index = int(len(data) * (1 - test_size))
+        else:
+            index = test_size
+
+        return data[:index], data[index:]
+
+    elif axis == 1:
+
+        if horizon == 0 or n == 0:
+            raise AttributeError("You need to provide non-zero `horizon` and `n` parameters when axis=1")
+
+        train_set = []
```

Comment:
This assumes that the datasets fit into memory.
Another option to scale to bigger datasets here could be to build our own Sequence implementation. Something like:
```
class MyTimeSeriesSequence(Sequence):
    def __init__(data):
        self.data = data

    def __getitem__(idx: int):
        ts = self.data[idx]
        ts_length = len(ts)
        ...
        return ts[:train_end_index], ts[test_start_index:]

...

return MyTimeSeriesSequence(data)
```

In this example it would make the return type of the method `Sequence[Tuple]` instead of `Tuple[Sequence]`, but it can be changed if `Tuple[Sequence]` makes more sense (wdyt about this?)

Reply:
How about there's going to be additional parameter where you can provide your own implementation of sequence that defaults to `list()`?

Reply:
(I have implemented this behavior)

Reply:
That would make it quite complex to use for users no? It means they would have to implement their own class. Whereas if you implement the class for them they don't have to worry about it. Is there a use-case where a class looking like the one I'm proposing above wouldn't work?

Reply:
Everything is fine with your implementation. I was always hesitant to implement things that may not be used at all, I'm a big fan of "less is more". I'm assuming, based on what you said about use cases, that in 90% of the cases `list()` will be completely sufficient. For the remaining 10% of the cases, where large datasets are used, I'm assuming these will be the power users that know how to implement it. I'm also a big proponent of flexibility with sane defaults. And I can imagine some exotic cases where custom implementation will be required (eg. storing data on the disk, using a joblib to manage memory or using tensors as an internal repr etc.).